AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/InitialSync/InitialSyncOrchestrator.swift

```diff
@@ -9,7 +9,7 @@ import Amplify
 import AWSPluginsCore
 
 protocol InitialSyncOrchestrator {
-    func sync(completion: @escaping (Result<Void, DataStoreError>) -> Void)
+    func sync(completion: @escaping (Result<ModelSyncedPayload?, DataStoreError>) -> Void)
```

Comment:
As noted above, you may be able to make the success case a non-optional `ModelSyncedPayload`

Reply:
But if I make it non-optional, line here should something back to caller (`RemoteSyncEngine`): https://github.com/aws-amplify/amplify-ios/blob/e6db87a8992926e8fabfcfd0c8953adc65d794b2/AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/InitialSync/InitialSyncOrchestrator.swift#L125
I think that's not I want since the caller (`RemoteSyncEngine` class doesn't care about `ModelSyncedPayload`). And that's why I make `ModelSyncedPayload` as an optional value to return.